seqr/views/utils/terra_api_utils.py

```diff
@@ -100,39 +111,48 @@ def _get_social_access_token(user):
     if (social.extra_data['auth_time'] + social.extra_data['expires'] - 10) <= int(
             time.time()):  # token expired or expiring?
         strategy = load_strategy()
+        logger.info('Refreshing access token', extra={'user': user})
         try:
             social.refresh_token(strategy)
         except Exception as ee:
-            logger.warning('Refresh token failed. {}'.format(str(ee)))
-            raise TerraAPIException('Refresh token failed. {}'.format(str(ee)), 401)
+            logger.warning(traceback.format_exc())
+            logger.warning('Refresh token failed. {}'.format(str(ee)), extra={'user': user})
+            raise TerraRefreshTokenFailedException('Refresh token failed. {}'.format(str(ee)))
     return social.extra_data['access_token']
 
 
-def anvil_call(method, path, access_token, user=None, headers=None, root_url=None, data=None):
+def anvil_call(method, path, access_token, user=None, headers=None, root_url=None, data=None, handle_errors=False):
     url, headers = _get_call_args(path, headers, root_url)
     request_func = getattr(requests, method)
     headers.update({'Authorization': 'Bearer {}'.format(access_token)})
     r = request_func(url, data=data, headers=headers)
 
+    exception = None
     if r.status_code == 404:
-        raise TerraNotFoundException('{} called Terra API: {} /{} got status 404 with reason: {}'
+        exception = TerraNotFoundException('{} called Terra API: {} /{} got status 404 with reason: {}'
                                      .format(user, method.upper(), path, r.reason))
-    if r.status_code == 403:
-        raise PermissionDenied('{} got access denied (403) from Terra API: {} /{} with reason: {}'
+    elif r.status_code == 403:
+        exception = PermissionDenied('{} got access denied (403) from Terra API: {} /{} with reason: {}'
                                .format(user, method.upper(), path, r.reason))
 
-    if r.status_code != 200:
-        raise TerraAPIException('Error: called Terra API: {} /{} got status: {} with a reason: {}'.format(method.upper(),
+    elif r.status_code != 200:
+        exception  = TerraAPIException('Error: called Terra API: {} /{} got status: {} with a reason: {}'.format(method.upper(),
             path, r.status_code, r.reason), r.status_code)
 
+    if exception:
+        if handle_errors:
+            logger.warning(str(exception))
+            return {}
```

Comment:
Since all types of exceptions will be handled, it has no reason to distinguish them. They can all belong to `status_code != 200`.

Reply:
I found we still need `TerraNotFoundException`.

Reply:
The exception for `status_code == 403` should be treated the same as `status_code != 200`.

Reply:
no, they format totally different error message